### PR TITLE
analytics(replay): analytics for hydration issue details and slider interactions

### PR DESCRIPTION
**Pull request by aliu39**
Ref https://github.com/getsentry/sentry/issues/74456

- Tracks first render of a hydrate error on issue details, after page load
- Tracks first click on the slider's divider bar, after the last render of the slider diff

Current behavior:
https://github.com/user-attachments/assets/0a6431d1-00a8-43fd-befb-0807d432518c

### CODE REVIEW

File: static/app/components/replays/diff/replaySliderDiff.tsx [hunk 1]
```diff
@@ -63,6 +66,23 @@ function DiffSides({leftOffsetMs, replay, rightOffsetMs, viewDimensions, width})
     },
   });
 
+  const organization = useOrganization();
+  const [dividerClicked, setDividerClicked] = useState(false); // never flips back to false
```
Comment (ryan953) [inline static/app/components/replays/diff/replaySliderDiff.tsx:70]: I would create this with `useRef` instead, right now it'll cause a re-render which isn't needed as we're just tracking the value not rendering anything.

```
  const dividerClickedRef = useRef(false);
  ...
  if (organization && !dividerClickedRef.current) {
    ...
    dividerClickedRef.current = true
```

**Comment by aliu39** [comment]
@ryan953 Got it. What are your thoughts on tracking first click after render vs after page load?

**Comment by ryan953** [comment]
> @ryan953 Got it. What are your thoughts on tracking first click after render vs after page load?

first click the way it is now, after render, seems fine. the person can use it a bunch of times through the day, we just don't need every single little mouse move

File: static/app/components/replays/diff/replaySliderDiff.tsx [hunk 2]
```diff
@@ -63,6 +66,23 @@ function DiffSides({leftOffsetMs, replay, rightOffsetMs, viewDimensions, width})
     },
   });
 
+  const organization = useOrganization();
+  const dividerClickedRef = useRef(false); // once set, never flips back to false
+
+  const onDividerMouseDownWithAnalytics: React.MouseEventHandler<HTMLElement> =
```
Comment (ryan953) [inline static/app/components/replays/diff/replaySliderDiff.tsx:72]: nice types!

File: static/app/components/replays/diff/replaySliderDiff.tsx [hunk 3]
```diff
@@ -44,7 +46,8 @@ export function ReplaySliderDiff({leftOffsetMs, replay, rightOffsetMs}: Props) {
 function DiffSides({leftOffsetMs, replay, rightOffsetMs, viewDimensions, width}) {
   const rightSideElem = useRef<HTMLDivElement>(null);
   const dividerElem = useRef<HTMLDivElement>(null);
-  const {onMouseDown} = useResizableDrawer({
+
+  const {onMouseDown: onDividerMouseDown} = useResizableDrawer({
```
Comment (ryan953) [inline static/app/components/replays/diff/replaySliderDiff.tsx:50]: that is a better name `onDividerMouseDown`

File: static/app/views/issueDetails/header.tsx [hunk 4]
```diff
@@ -83,6 +84,12 @@ function GroupHeaderTabs({
     group_has_replay: (replaysCount ?? 0) > 0,
   });
 
+  useEffect(() => {
+    if (organization && group.issueType === IssueType.REPLAY_HYDRATION_ERROR) {
+      trackAnalytics('replay.hydration-error.issue-details', {organization});
+    }
```
Comment (ryan953) [inline static/app/views/issueDetails/header.tsx:90]: typescript thinks that organization will always be set here, so we should be good without it in the check
```suggestion
    if (group.issueType === IssueType.REPLAY_HYDRATION_ERROR) {
      trackAnalytics('replay.hydration-error.issue-details', {organization});
    }
```

**Review by ryan953 — APPROVED** [review]

**Pull request merged**